pkg/pool-weighted/contracts/lib/CircuitBreakerLib.sol

```diff
@@ -312,6 +312,42 @@ library CircuitBreakerLib {
             );
     }
 
+    /**
+     * @notice Update the cached ratios given a new weight complement.
+     * @dev This might be used when weights are adjusted, pre-emptively updating the cache to improve performance
+     * of operations after the weight change completed. Note that this does not update the BPT price: this is still
+     * relative to the last call to `setCircuitBreakerFields`. The intent is only to optimize the automatic bounds
+     * adjustments due to changing weights.
+     */
+    function updateBoundRatios(bytes32 circuitBreakerState, uint256 weightComplement) internal pure returns (bytes32) {
```

Comment:
We could save some bytecode by using this function in `setCircuitBreakerFields` but I don't think that's required atm. We can revisit later if necessary.